src-tauri/src/commands.rs

```diff
@@ -386,7 +407,10 @@ fn get_aggregation(from: NaiveDateTime) -> Result<DateTimeAggregation, Error> {
     // Use hourly aggregation for longer periods
     let aggregation = match Utc::now().naive_utc() - from {
         duration if duration >= Duration::hours(8) => Ok(DateTimeAggregation::Hour),
-        duration if duration < Duration::zero() => Err(Error::InternalError),
+        duration if duration < Duration::zero() => Err(Error::InternalError(format!(
+            "Negative duration between dates: now and {}",
```

Comment:
Let's log `Utc::now().naive_utc()` too since this kind of error may be a result of incorrect system date.